### PR TITLE
Fix validate's executable dependency

### PR DESCRIPTION
**Pull request by izgzhen**
Addresses https://github.com/snowleopard/hadrian/issues/366

### CODE REVIEW

**Comment by izgzhen** [comment]
The problem can only be reproduced by invoking `validate` only without building top-level target first.

File: src/Rules/Test.hs [hunk 1]
```diff
@@ -22,6 +22,8 @@ testRules = do
         needBuilder $ Ghc CompileHs Stage2
         needBuilder $ GhcPkg Update Stage1
         needBuilder Hpc
+        needBuilder Hsc2Hs
+        need ["inplace/bin/hp2ps"]
```
Comment (snowleopard) [inline src/Rules/Test.hs]: Can we make this less fragile and compute the path? Alternatively, add a TODO to address this later.

**Comment by snowleopard** [comment]
By the way, there are a few other instances where we `need` filenames directly, so I'm planning to go through the code and solve this issue in general. So leaving a `TODO` is fine.

**Comment by izgzhen** [comment]
> By the way, there are a few other instances where we need filenames directly, so I'm planning to go through the code and solve this issue in general. So leaving a TODO is fine.

I will open an issue and leave a TODO. I think this type of task is relatively easy, well-defined, and not so urgent, maybe you don't have to do it yourself. Leave it for the first-time contributors, if any, would be nice I think.

**Comment by snowleopard** [comment]
@izgzhen Yes, I agree. Thank you!

**Pull request merged**